frontend/src/actions/failures/details.js

```diff
@@ -0,0 +1,42 @@
+import API, { parseResponseError } from 'api'
+import { addFlashMessage } from 'actions/flash-messages'
+
+import {
+  REQUEST_FAILURE_DETAILS,
+  RECEIVE_FAILURE_DETAILS,
+  REQUEST_FAILURE_DETAILS_FAILED
+} from 'actions'
+
+const requestFailureDetails = (failure) => ({
+  type: REQUEST_FAILURE_DETAILS,
+  failure
+})
+
+const receiveFailureDetails = (failure) => ({
+  type: RECEIVE_FAILURE_DETAILS,
+  failure
+})
+
+const requestFailureDetailsFailed = (failure, error) => ({
+  type: REQUEST_FAILURE_DETAILS_FAILED,
+  failure,
+  error
+})
+
+export const fetchFailureDetails = (failure) => (dispatch, getState) => {
+  dispatch(requestFailureDetails(failure))
+
+  return API.Failure
+    .findById({ id: failure.id })
+    .then((response) => dispatch(receiveFailureDetails(response.data)))
+    .catch((response) => {
+      const error = parseResponseError(response)
+      return Promise
+        .resolve()
+        .then(() => dispatch(requestFailureDetailsFailed(failure, error.message)))
```

Comment:
do we need to wrap this and the next dispatch in the promise?
couldn't we simply call one and then the other and return nothing for the catch block?